sdk/core/azure-core/azure/core/pipeline/_tools.py

```diff
@@ -32,3 +32,40 @@ def await_result(func, *args, **kwargs):
             "Policy {} returned awaitable object in non-async pipeline.".format(func)
         )
     return result
+
+def to_rest_request(pipeline_transport_request):
+    from ..rest import HttpRequest as RestHttpRequest
+    return RestHttpRequest(
+        method=pipeline_transport_request.method,
+        url=pipeline_transport_request.url,
+        headers=pipeline_transport_request.headers,
+        files=pipeline_transport_request.files,
+        data=pipeline_transport_request.data
+    )
+
+def to_rest_response(pipeline_transport_response):
```

Comment:
What will this method do if users have created their client with a custom transport?
Does it matter here?

Reply:
if they create their client with a custom transport, since i don't have a corresponding rest transport response to change it to, I change it to the default `azure.core.rest.HttpResponse` or `azure.core.rest.AsyncHttpResponse`